src/domain/datasets/domain/src/services/dataset_entry_service.rs

```diff
@@ -12,12 +12,18 @@ use internal_error::InternalError;
 use opendatafabric as odf;
 use thiserror::Error;
 
-use crate::DatasetEntry;
+use crate::{DatasetEntry, DatasetEntryStream};
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 #[async_trait::async_trait]
 pub trait DatasetEntryService: Sync + Send {
+    // TODO: Private Datasets: tests
+    // TODO: Private Datasets: extract to DatasetEntryRegistry?
```

Comment:
There is no such thing as `DatasetEntryRegistry`. If you mean `DatasetRegistry`, it should not contain these operations, it plays a different role. 

I think these operations should stay here.

Reply:
> There is no such thing as `DatasetEntryRegistry`

I was thinking of adding the thing :) ... 

> I think these operations should stay here.

... but I agree we should just leave it here.

Reply:
TODOs have been updated

Done